src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -229,6 +229,34 @@ class E
             comp.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(49628, "https://github.com/dotnet/roslyn/issues/49628")]
+        public void AmbigCtor()
+        {
+            var src = @"
+record R(R x);
```

Comment:
From what I understand, this change is specific to positional records only.

Shouldn't the same error be issued for something like the following:

```csharp
record R(string S)
{
    public R(R x) {}
};
```

Reply:
> Shouldn't the same error be issued for something like the following:

Is there a problem with this scenario?

---
In reply to: [543321591](https://github.com/dotnet/roslyn/pull/49975#discussion_r543321591) [](ancestors = 543321591)

Reply:
> Is there a problem with this scenario?

@AlekseyTs It's *currently* allowed in [master](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4LgYwPYICYAIBKAFAAICMADLgMoCUAsAFADejubuxAzAYfrgB41cTAL6MRAbkaMAdhDBwoABwho4HUgDp8AVxkwAlgs0BhDGCUGANoiqIAbgbVQmxAEy4AklACi/eAhyVp4yBjCiIkA).

The PR doesn't seem to disallow it.
So my question was: should this scenario (explicit declaration of a copy constructor) be disallowed?

Reply:
> So my question was: should this scenario (explicit declaration of a copy constructor) be disallowed?

Does this violate specification?

---
In reply to: [543402442](https://github.com/dotnet/roslyn/pull/49975#discussion_r543402442) [](ancestors = 543402442)

Reply:
> Does this violate specification?

On its current state, I think no.

> What would be the reason to do this?

It's still ambiguous (if I'm not missing something). Also, if it will replace the synthesized copy constructor, probably `with` initializers won't work as expected in this case (because the `<Clone>$` method depends on the copy constructor). So at minimum I'd expect either a compiler warning or an IDE suggestion (or a CA code quality rule) to warn for this.

Reply:
> It's still ambiguous. 

With what?

> (if I'm not missing something)

This PR is about detecting an ambiguity between a synthesized copy constructor and a primary constructor. This ambiguity for explicitly declared copy constructor is already detected, I think.

> So at minimum I'd expect either a compiler warning or an IDE suggestion (or a CA code quality rule) to warn for this.

Explicit declaration of a copy constructor is intentionally supported/allowed ([https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/records.md#copy-and-clone-members](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/records.md#copy-and-clone-members)).

---
In reply to: [543424682](https://github.com/dotnet/roslyn/pull/49975#discussion_r543424682) [](ancestors = 543424682)